src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -441,16 +438,18 @@ func (s *commitLogSource) Read(
 				tagIter = tagDecoder
 			}
 
-			// Check out the series for writing.
 			series, err := accumulator.CheckoutSeries(entry.Series.ID, tagIter)
 			if err != nil {
 				return bootstrap.NamespaceResults{}, err
 			}
 
+			series.Shard = entry.Series.Shard
```

Comment:
Could we make `CheckoutSeries(...)` perhaps return the shard?  Might make this less dependent on the caller to do the right thing here.

Reply:
Yeah, that's probably better (although it would need to take in shard too since that's not available on the current args), will look into it after finishing the rest of the pr and getting it into a working state